tests/unittests/tests-tsrb/tests-tsrb.c

```diff
@@ -115,6 +115,43 @@ static void test_get(void)
     }
 }
 
+static void test_peek_one(void)
+{
+    int res;
+
+    TEST_ASSERT_EQUAL_INT(-1, tsrb_peek_one(&_tsrb));
+    TEST_ASSERT_EQUAL_INT(0, tsrb_add_one(&_tsrb, TEST_INPUT));
+    TEST_ASSERT_EQUAL_INT(TEST_INPUT, tsrb_peek_one(&_tsrb));
+    TEST_ASSERT_EQUAL_INT(TEST_INPUT, tsrb_peek_one(&_tsrb));
+    TEST_ASSERT_EQUAL_INT(TEST_INPUT, tsrb_get_one(&_tsrb));
+    TEST_ASSERT_EQUAL_INT(-1, tsrb_peek_one(&_tsrb));
+    TEST_ASSERT_EQUAL_INT(0, tsrb_add_one(&_tsrb, 0xff));
+    res = tsrb_peek_one(&_tsrb);
+    TEST_ASSERT_EQUAL_INT(0xff, res);
+    /* 0xff is -1 in signed int8_t */
+    TEST_ASSERT(-1 != res);
+}
+
+static void test_peek(void)
+{
+    TEST_ASSERT(BUFFER_SIZE < sizeof(_io_buffer));
+    TEST_ASSERT_EQUAL_INT(0, tsrb_peek(&_tsrb, _io_buffer,
+                                      sizeof(_io_buffer)));
+
+    for (int i = 0; i < BUFFER_SIZE; i++) {
+        TEST_ASSERT_EQUAL_INT(0, tsrb_add_one(&_tsrb, TEST_INPUT + i));
+    }
+    TEST_ASSERT_EQUAL_INT(BUFFER_SIZE, tsrb_peek(&_tsrb, _io_buffer,
+                                                sizeof(_io_buffer)));
+    for (int i = 0; i < BUFFER_SIZE; i++) {
+        TEST_ASSERT_EQUAL_INT((TEST_INPUT + i), _io_buffer[i]);
+    }
+    for (int i = BUFFER_SIZE; i < (int)sizeof(_io_buffer); i++) {
+        TEST_ASSERT_EQUAL_INT(IO_BUFFER_CANARY, _io_buffer[i]);
+    }
```

Comment:
Maybe use `unsigned` ?

```suggestion
    for (unsigned i = 0; i < BUFFER_SIZE; i++) {
        TEST_ASSERT_EQUAL_INT(0, tsrb_add_one(&_tsrb, TEST_INPUT + i));
    }
    TEST_ASSERT_EQUAL_INT(BUFFER_SIZE, tsrb_peek(&_tsrb, _io_buffer,
                                                sizeof(_io_buffer)));
    for (unsigned i = 0; i < BUFFER_SIZE; i++) {
        TEST_ASSERT_EQUAL_INT((TEST_INPUT + i), _io_buffer[i]);
    }
    for (unsigned i = BUFFER_SIZE; i < sizeof(_io_buffer); i++) {
        TEST_ASSERT_EQUAL_INT(IO_BUFFER_CANARY, _io_buffer[i]);
    }
```

Reply:
I mostly copied over the other tests, so if this was to be changed I would do it for the others as well...